src/Polling/Frequent/FrequentPollingActivity.cs

```diff
@@ -0,0 +1,44 @@
+using Microsoft.Extensions.Logging;
+using Temporalio.Activities;
+using Temporalio.Exceptions;
+using TemporalioSamples.Polling.Common;
+
+namespace TemporalioSamples.Polling.Frequent;
+
+public class FrequentPollingActivity : IPollingActivity
+{
+    private static int pollDurationSections = 1;
+
+    private readonly TestService service;
+
+    public FrequentPollingActivity(TestService service) => this.service = service;
+
+    [Activity]
+    public async Task<string> DoPollAsync()
+    {
+        while (true)
+        {
+            try
+            {
+                return await service.GetServiceResultAsync();
```

Comment:
```suggestion
                return await service.GetServiceResultAsync(ActivityExecutionContext.Current.CancellationToken);
```

And make the service code accept it